### PR TITLE
ref(authentication): handle joining in low bandwidth mode

### PR DESCRIPTION
**Pull request by Calinteodor**
Tested multiple cases, check individual commits.

### CODE REVIEW

File: react/features/app/actions.native.ts [hunk 1]
```diff
@@ -155,6 +157,14 @@ export function appNavigate(uri?: string, options: IReloadNowOptions = {}) {
         dispatch(setRoom(room));
 
         if (!room) {
+            const { startAudioOnly } = getState()['features/base/settings'];
+
+            if (isTokenAuthEnabled(config) && startAudioOnly) {
+                dispatch(updateSettings({
+                    startAudioOnly: false
```
Comment (saghul) [inline react/features/app/actions.native.ts:164]: I don't understand what this is fixing.

File: react/features/conference/components/native/Conference.tsx [hunk 2]
```diff
@@ -230,14 +230,20 @@ class Conference extends AbstractConference<IProps, State> {
...
+            _showLobby,
+            _startCarMode
         } = this.props;
 
         if (!prevProps._showLobby && _showLobby) {
             navigate(screen.lobby.root);
         }
 
         if (prevProps._showLobby && !_showLobby) {
+            if (_audioOnlyEnabled && _startCarMode) {
```
Comment (saghul) [inline react/features/conference/components/native/Conference.tsx:243]: This is so we remain in car mode right?

File: react/features/prejoin/components/native/Prejoin.tsx [hunk 3]
```diff
@@ -97,6 +99,14 @@ const Prejoin: React.FC<IPrejoinProps> = ({ navigation }: IPrejoinProps) => {
     }, [ dispatch, hasDisplayName, isDisplayNameMissing, onJoin ]);
 
     const onJoinLowBandwidth = useCallback(() => {
+
+        // Authentication navigation to the external url resets the state
+        if (isTokenAuthEnabled(config)) {
```
Comment (saghul) [inline react/features/prejoin/components/native/Prejoin.tsx:104]: This is kinda a workaround. The auth service should retain the startAudioOnly parameter. @damencho does it do that now? Will it retain any hash parameters we pass?

**Review by saghul — DISMISSED** [review]

File: react/features/app/actions.native.ts (hunk 1 above)
Comment (Calinteodor) [reply react/features/app/actions.native.ts:164]: It resets the start audio only state.

File: react/features/prejoin/components/native/Prejoin.tsx (hunk 3 above)
Comment (Calinteodor) [reply react/features/prejoin/components/native/Prejoin.tsx]: @saghul It does retain. You were right. I updated the PR and tested it.

File: react/features/app/actions.any.ts [hunk 4]
```diff
@@ -120,7 +122,7 @@ export function maybeRedirectToTokenAuthUrl(
         const room = state['features/base/conference'].room;
         const { tenant } = parseURIString(locationURL.href) || {};
 
-        getTokenAuthUrl(config, room, tenant, true, locationURL)
+        getTokenAuthUrl((audioOnlyEnabled || startAudioOnly), config, room, tenant, true, locationURL)
```
Comment (Calinteodor) [inline react/features/app/actions.any.ts:125]: @saghul if reloadNow is dispatched and the participant joined low bandwidth mode or config startAudioOnly was set to true in the first place, this is expected, right? To rejoin on the same terms.

**Review by saghul — APPROVED** [review]

**Pull request merged**